cmake/FindSQLite3.cmake

```
@@ -59,5 +59,6 @@ endif()
 
 # Workaround for https://github.com/gazebosim/gz-cmake/issues/312
 if(TARGET SQLite3::SQLite3 AND NOT TARGET SQLite::SQLite3)
-  add_library(SQLite::SQLite3 ALIAS SQLite3::SQLite3)
+  add_library(SQLite::SQLite3 INTERFACE)
+  target_link_libraries(SQLite::SQLite3 INTERFACE SQLite3::SQLite3)
```

Based on https://discourse.cmake.org/t/alias-and-not-globally-visible-targets/6305/2, I tried this and it seems to work
```suggestion
  add_library(SQLite::SQLite3 INTERFACE IMPORTED)
  # Equivalent to target_link_libraries INTERFACE, but compatible with CMake 3.10
  set_target_properties(SQLite::SQLite3 PROPERTIES INTERFACE_LINK_LIBRARIES SQLite3::SQLite3)
```

---

I was looking at this same issue. Need to declare the `IMPORTED` keyword yes.